dtschema/lib.py

```diff
@@ -358,6 +358,10 @@ def fixup_node_props(schema):
 
     schema['properties']['phandle'] = True
     schema['properties']['status'] = True
+    schema['properties']['pinctrl-names'] = True
```

Comment:
I don't we should handle it this way as individual bindings should document whether or not they use pinctrl properties and what the names are.

Reply:
It's true that this would override everything if a driver has a particular requirement.

But at the same time, the core also grabs the pinctrl default state for any device, so any device can use or not those properties, without any particular constraints.

Even just getting the default pinctrl property is quite tricky: the trivial way to do this is to use pinctrl-names = "default" and using pinctrl-0, but having multiple "default" pinctrl-names with the same number of pinctrl-* properties is just as valid.

I really feel like this will lead to a lot of boilerplate, with much of the bindings getting this wrong.

Maybe we should just avoid erasing any pinctrl properties if they exist, and add those defaults if they don't?

Reply:
Okay, if we don't override any existing properties, that's fine with me.

Reply:
I just pushed a new version that checks for pinctrl related property in a schema before adding them. Let me know if it's better. Thanks!